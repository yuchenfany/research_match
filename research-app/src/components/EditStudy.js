/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import Select from 'react-select';
import { useNavigate } from 'react-router-dom';
import '../assets/index.css';
// study, setStudy,
function EditStudy({ user, study, setStudy }) {
  // async function getStudy() {
  //   const studyData = await fetch(`http://localhost:5000/study/${study.studyId}`, {
  //     method: 'GET',
  //     headers: {
  //       'Content-Type': 'application/json',
  //     },
  //   });
  //   const data = await studyData.json();

  //   return data;
  // }
  const navigate = useNavigate();
  const Tags = [
    { label: 'Diabetes', value: 'diabetes' },
    { label: 'Cancer', value: 'cancer' },
    { label: 'Social', value: 'social' },
    { label: 'placebo', value: 'placebo' },
    { label: 'Brain', value: 'brain' },
    { label: 'Physical', value: 'physical' },
  ];
  const customStyles = {
    menu: (provided, state) => ({
      ...provided,
      width: 'state.selectProps.width,',
      color: state.selectProps.menuColor,
    }),
    control: (provided, state) => ({
      ...provided,
      background: '#F9FFFE',
      borderColor: '#808A8F',
      fontSize: '14px',
      boxShadow: state.isFocused ? null : null,
    }),
    multiValueLabel: (styles) => ({
      ...styles,
      backgroundColor: '#BBEFEB',
    }),
    multiValueRemove: (styles) => ({
      ...styles,
      backgroundColor: '#BBEFEB',
      ':hover': {
        color: '#4aa8a2',
      },
    }),
  };
  // async function getNextStudyID() {
  //   const studyData = await fetch('http://localhost:5000/findMax', {
  //     method: 'GET',
  //     headers: {
  //       'Content-Type': 'application/json',
  //     },
  //   });
  //   const data = await studyData.json();
  //   return data;
  // }

  // async function addStudy() {
  //   await fetch('http://localhost:5000/add-study', {
  //     method: 'POST',
  //     headers: {
  //       'Content-Type': 'application/json',
  //     },
  //     body: JSON.stringify(study),
  //   })
  //     .catch((e) => {
  //       window.alert(e);
  //     });

  // }
  async function verify() {
    setStudy({
      title: study.title,
      description: study.description,
      compensation: study.compensation,
      duration: study.duration,
      tags: study.tags,
      participants: study.participants,
      studyId: study.studyId,
      researchers: study.researchers,
    // }, () => {
    //   addStudy();
    });
    const myobj = {
      title: study.title,
      description: study.description,
      compensation: study.compensation,
      duration: study.duration,
      tags: study.tags,
      participants: study.participants,
      studyId: study.studyId,
      researchers: study.researchers,
    };
    // edits the study
    await fetch('http://localhost:5000/study/edit-study', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(myobj),
    })
      .catch((e) => {
        window.alert(e);
      });
    return true;
  }

  const updateTitle = async (event) => {
    setStudy({
      title: event.target.value,
      description: study.description,
      compensation: study.compensation,
      duration: study.duration,
      tags: study.tags,
      participants: study.participants,
      studyId: study.studyId,
      researchers: study.researchers,
    });
  };
  const updateDescription = async (event) => {
    setStudy({
      title: study.title,
      description: event.target.value,
      compensation: study.compensation,
      duration: study.duration,
      tags: study.tags,
      participants: study.participants,
      studyId: study.studyId,
      researchers: study.researchers,
    });
  };
  const updateCompensation = async (event) => {
    setStudy({
      title: study.title,
      description: study.description,
      compensation: event.target.value,
      duration: study.duration,
      tags: study.tags,
      participants: study.participants,
      studyId: study.studyId,
      researchers: study.researchers,
    });
  };
  const updateDuration = async (event) => {
    setStudy({
      title: study.title,
      description: study.description,
      compensation: study.compensation,
      duration: event.target.value,
      tags: study.tags,
      participants: study.participants,
      studyId: study.studyId,
      researchers: study.researchers,
    });
  };
  const updateResearcher = async (event) => {
    setStudy({
      title: study.title,
      description: study.description,
      compensation: study.compensation,
      duration: study.duration,
      tags: study.tags,
      participants: study.participants,
      studyId: study.studyId,
      researchers: event.target.value,
    });
  };
  async function getTagsArr(tags) {
    const arr = [];

    for (let i = 0; i < tags.length; i += 1) {
      arr.push(tags[i].value);
    }

    return arr;
  }

  const updateTags = async (tags) => {
    const arr = await getTagsArr(tags);
    setStudy({
      title: study.title,
      description: study.description,
      compensation: study.compensation,
      duration: study.duration,
      tags: arr,
      participants: study.participants,
      studyId: study.studyId,
      researchers: study.researchers,
    });
  };
  async function handleSubmit(event) {
    if (await verify()) {
      navigate('/researcher-home');
    } else {
      event.preventDefault();
    }
  }

  // deletes a study
  async function deleteStudy() {
    await fetch(`http://localhost:5000/study/${study.studyId}`, {
      method: 'DELETE',
      body: null,
    });
    navigate('/researcher-home');
  }

  // for deleting a study: updates a researcher's study array
  async function updateResearcherStudies() {
    const updatedStudies = user.studies.filter((e) => e !== study.studyId);

    const bodyObj = {
      username: user.username,
      password: user.password,
      name: user.name,
      organization: user.organization,
      studies: updatedStudies,
      type: user.type,
      title: user.title,
    };

    await fetch(`http://localhost:5000/record/researcher-studies/${user.username}`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(bodyObj),
    })
      .catch((e) => {
        window.alert(e);
      });
    return true;
  }

  // calling delete study functions
  async function handleDelete() {
    deleteStudy().then(updateResearcherStudies());
  }

  async function getParticipants() {
    const response = await fetch(`http://localhost:5000/study/${study.studyId}`, {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
      },
    }).catch((e) => { window.alert(e); });
    const json = await response.json();
    const participantIds = json?.participants ?? [];
    const participants = await Promise.all(
      participantIds.map(
        async (id) => {
          const participantData = await fetch(`http://localhost:5000/record/${id}`, {
            method: 'GET',
            headers: {
              'Content-Type': 'application/json',
            },
          }).catch((e) => { window.alert(e); });
          const participant = await participantData.json();
          return participant;
        },
      ),
    );
    return participants ?? [];
  }

  async function removeStudyForParticipants(participants) {
    await Promise.all(
      participants.map(
        async (participant) => {
          if (!participant) {
            return;
          }
          const updatedStudies = participant.enrolled.filter((e) => e !== study.studyId) ?? [];
          const bodyObj = participant;
          bodyObj.enrolled = updatedStudies;
          await fetch(`http://localhost:5000/record/participant-edit/${participant.username}`, {
            method: 'POST',
            body: JSON.stringify(bodyObj),
            headers: {
              'Content-Type': 'application/json',
            },
          }).catch((e) => { window.alert(e); });
        },
      ),
    );
  }

  async function handleClose() {
    await getParticipants().then((participants) => { removeStudyForParticipants(participants); });
    const bodyObj = study;
    bodyObj.closed = '1';
    bodyObj.participants = [];
    await fetch('http://localhost:5000/add-study', {
      method: 'POST',
      body: JSON.stringify(bodyObj),
      headers: {
        'Content-Type': 'application/json',
      },
    }).catch((e) => { window.alert(e); });
    navigate('/researcher-home');
  }

  return (
    <div className="Profile">
      <div className="profile-flex">
        <div className="header-left"> Edit Study </div>
        <div className="title-row">
          <div>Title</div>
          <input
            className="input-field"
            type="text"
            id="title"
            value={study.title}
            onChange={updateTitle}
          />
        </div>
        <div className="description-row">
          <div>Description</div>
          <input
            className="input-field"
            type="text"
            id="description"
            value={study.description}
            onChange={updateDescription}
          />
          <div>Compensation</div>
          <input
            className="input-field"
            type="text"
            id="compensation"
            value={study.compensation}
            onChange={updateCompensation}
          />
          <div>Duration</div>
          <input
            className="input-field"
            type="text"
            id="duration"
            value={study.duration}
            onChange={updateDuration}
          />
          <div>Lead Researcher</div>
          <input
            className="input-field"
            type="text"
            id="researchers"
            value={study.researchers}
            onChange={updateResearcher}
          />
        </div>
        <div className="profile-row">
          <div>Tags</div>
        </div>
        <div className="profile-row">
          <Select
            options={Tags}
            isMulti
            onChange={(tags) => updateTags(tags)}
            value={study.tags}
            styles={customStyles}
          />
        </div>
        <input className="signup-button" type="submit" value="Update" onClick={handleSubmit} />
        <input className="signup-button" type="button" value="Delete Study" onClick={handleDelete} />
        <input className="signup-button" type="button" value="Close Study" onClick={handleClose} />
      </div>
    </div>
  );
}

export default EditStudy;
