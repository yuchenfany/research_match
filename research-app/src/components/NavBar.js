/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-filename-extension */
import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import '../assets/index.css';

function NavBar({ user, setUser }) {
  const [accountToggled, setAccountToggled] = useState(false);
  const navigate = useNavigate();
  const editProfile = () => {
    let arr = {};

    if (user.type === 0) { // participant
      arr = {
        username: user.username,
        password: user.password,
        enrolled: user.enrolled,
        age: user.age,
        heightFeet: user.heightFeet,
        heightInches: user.heightInches,
        weight: user.weight,
        sex: user.sex,
        gender: user.gender,
        allergies: user.allergies,
        phys: user.phys,
        psych: user.psych,
        med: user.med,
        type: user.type,
      };
    } else if (user.type === 1) { // researcher
      arr = {
        username: user.username,
        password: user.password,
        name: user.name,
        organization: user.organization,
        type: user.type,
      };
    }

    // store current state in case of 'Cancel'
    localStorage.setItem(user.username, JSON.stringify(arr));
    navigate(`/${(user.type ?? 0) === 0 ? 'participant' : 'researcher'}-edit`);
  };

  const accountOptions = (
    <div
      className="acct-options"
      style={{ display: accountToggled ? 'flex' : 'none' }}
    >
      <button
        type="button"
        className="account-btn"
        onClick={editProfile}
      >
        EDIT PROFILE
      </button>
      <button
        type="button"
        className="account-btn"
        onClick={async () => { navigate('/delete-account', { user, setUser }); }}
      >
        DELETE ACCOUNT
      </button>
      <button
        type="button"
        className="logout-btn"
        onClick={async () => { navigate('/', { user, setUser }); }}
      >
        LOGOUT
      </button>
    </div>
  );

  const toggleAccountOptions = () => {
    setAccountToggled(!accountToggled);
  };

  return (
    <div className="nav">
      <button
        type="button"
        className="nav-btn"
        onClick={async () => {
          navigate(`/${(user.type ?? 0) === 0 ? 'participant' : 'researcher'}-home`);
        }}
      >
        HOME
      </button>
      <button
        type="button"
        className="nav-btn"
        onClick={async () => {
          navigate('/display-studies');
        }}
      >
        STUDIES
      </button>
      <button type="button" className="nav-btn">MESSAGES</button>
      <button
        type="button"
        className="nav-btn"
        onClick={async () => { navigate('/dashboard', { user }); }}
      >
        ANALYTICS
      </button>
      <button type="button" className="account-btn" onClick={toggleAccountOptions}>ACCOUNT</button>
      {accountOptions}
    </div>
  );
}

export default NavBar;
