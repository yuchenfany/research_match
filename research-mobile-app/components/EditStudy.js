/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-filename-extension */
import React, { useState } from 'react';
import { View, Text, Button } from 'react-native';
// import Select from 'react-select';

function EditStudy({ route, navigation }) {
  let { user, setUser, study, setStudy } = route.params;

  const [temp, setTemp] = useState({
    _id: study._id,
    title: study.title,
    description: study.description,
    compensation: study.compensation,
    duration: study.duration,
    tags: study.tags,
    participants: study.participants,
    studyId: study.studyId,
    researchers: study.researchers,
  });

  // async function getStudy() {
  //   const studyData = await fetch(`http://localhost:5000/study/${study.studyId}`, {
  //     method: 'GET',
  //     headers: {
  //       'Content-Type': 'application/json',
  //     },
  //   });
  //   const data = await studyData.json();

  //   return data;
  // }

  const Tags = [
    { label: 'Diabetes', value: 'diabetes' },
    { label: 'Cancer', value: 'cancer' },
    { label: 'Social', value: 'social' },
    { label: 'placebo', value: 'placebo' },
    { label: 'Brain', value: 'brain' },
    { label: 'Physical', value: 'physical' },
  ];
  const customStyles = {
    menu: (provided, state) => ({
      ...provided,
      width: 'state.selectProps.width,',
      color: state.selectProps.menuColor,
    }),
    control: (provided, state) => ({
      ...provided,
      background: '#F9FFFE',
      borderColor: '#808A8F',
      fontSize: '14px',
      boxShadow: state.isFocused ? null : null,
    }),
    multiValueLabel: (styles) => ({
      ...styles,
      backgroundColor: '#BBEFEB',
    }),
    multiValueRemove: (styles) => ({
      ...styles,
      backgroundColor: '#BBEFEB',
      ':hover': {
        color: '#4aa8a2',
      },
    }),
  };
  // async function getNextStudyID() {
  //   const studyData = await fetch('http://localhost:5000/findMax', {
  //     method: 'GET',
  //     headers: {
  //       'Content-Type': 'application/json',
  //     },
  //   });
  //   const data = await studyData.json();
  //   return data;
  // }

  // async function addStudy() {
  //   await fetch('http://localhost:5000/add-study', {
  //     method: 'POST',
  //     headers: {
  //       'Content-Type': 'application/json',
  //     },
  //     body: JSON.stringify(study),
  //   })
  //     .catch((e) => {
  //       window.alert(e);
  //     });

  // }
  async function verify() {
    // const myobj = {
    //   title: temp.title,
    //   description: temp.description,
    //   compensation: temp.compensation,
    //   duration: temp.duration,
    //   tags: temp.tags,
    //   participants: temp.participants,
    //   studyId: temp.studyId,
    //   researchers: temp.researchers,
    // };
    // console.log(myobj.title);

    // edits the study
    await fetch('http://localhost:5000/study/edit-study', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(temp),
    })
      .catch((e) => {
        window.alert(e);
      });
    return true;
  }

  const updateTitle = async (event) => {
    await setTemp({
      _id: temp._id,
      title: event.target.value,
      description: temp.description,
      compensation: temp.compensation,
      duration: temp.duration,
      tags: temp.tags,
      participants: temp.participants,
      studyId: temp.studyId,
      researchers: temp.researchers,
    });
  };
  const updateDescription = async (event) => {
    await setTemp({
      _id: temp._id,
      title: temp.title,
      description: event.target.value,
      compensation: temp.compensation,
      duration: temp.duration,
      tags: temp.tags,
      participants: temp.participants,
      studyId: temp.studyId,
      researchers: temp.researchers,
    });
  };
  const updateCompensation = async (event) => {
    await setTemp({
      _id: temp._id,
      title: temp.title,
      description: temp.description,
      compensation: event.target.value,
      duration: temp.duration,
      tags: temp.tags,
      participants: temp.participants,
      studyId: temp.studyId,
      researchers: temp.researchers,
    });
  };
  const updateDuration = async (event) => {
    await setTemp({
      _id: temp._id,
      title: temp.title,
      description: temp.description,
      compensation: temp.compensation,
      duration: event.target.value,
      tags: temp.tags,
      participants: temp.participants,
      studyId: temp.studyId,
      researchers: temp.researchers,
    });
  };
  const updateResearcher = async (event) => {
    await setTemp({
      _id: temp._id,
      title: temp.title,
      description: temp.description,
      compensation: temp.compensation,
      duration: temp.duration,
      tags: temp.tags,
      participants: temp.participants,
      studyId: temp.studyId,
      researchers: event.target.value,
    });
  };
  async function getTagsArr(tags) {
    const arr = [];

    for (let i = 0; i < tags.length; i += 1) {
      arr.push(tags[i].value);
    }

    return arr;
  }

  const updateTags = async (tags) => {
    const arr = await getTagsArr(tags);
    setStudy({
      title: study.title,
      description: study.description,
      compensation: study.compensation,
      duration: study.duration,
      tags: arr,
      participants: study.participants,
      studyId: study.studyId,
      researchers: study.researchers,
    });
  };

  async function handleSubmit(event) {
    if (await verify()) {
      await setStudy(temp);
      navigation.navigate('ResearcherHome', {
        user,
        setUser,
        setStudy,
      });
    } else {
      event.preventDefault();
    }
  }

  // deletes a study
  async function deleteStudy() {
    await fetch(`http://localhost:5000/study/${study.studyId}`, {
      method: 'DELETE',
      body: null,
    });
    navigation.navigate('ResearcherHome', {
      user,
      setUser,
      setStudy,
    });
  }

  // finding researcher's list of studies
  async function getStudyIds() {
    const data = await fetch(`http://localhost:5000/record/${user.username}`, {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
      },
    });
    const json = await data.json();
    return json?.studies ?? [];
  }

  // for deleting a study: updates a researcher's study array
  async function updateResearcherStudies() {
    const currStudies = await getStudyIds();
    const updatedStudies = currStudies.filter((e) => e !== study.studyId);

    const bodyObj = {
      username: user.username,
      studies: updatedStudies,
    };

    await fetch(`http://localhost:5000/record/researcher-studies/${user.username}`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(bodyObj),
    })
      .catch((e) => {
        window.alert(e);
      });
    return true;
  }

  // calling delete study functions
  async function handleDelete() {
    deleteStudy().then(updateResearcherStudies());
  }

  // async function getParticipants() {
  //   const response = await fetch(`http://localhost:5000/study/${study.studyId}`, {
  //     method: 'GET',
  //     headers: {
  //       'Content-Type': 'application/json',
  //     },
  //   }).catch((e) => { window.alert(e); });
  //   const json = await response.json();
  //   const participantIds = json?.participants ?? [];
  //   const participants = await Promise.all(
  //     participantIds.map(
  //       async (id) => {
  //         const participantData = await fetch(`http://localhost:5000/record/${id}`, {
  //           method: 'GET',
  //           headers: {
  //             'Content-Type': 'application/json',
  //           },
  //         }).catch((e) => { window.alert(e); });
  //         const participant = await participantData.json();
  //         return participant;
  //       },
  //     ),
  //   );
  //   return participants ?? [];
  // }

  // async function removeStudyForParticipants(participants) {
  //   await Promise.all(
  //     participants.map(
  //       async (participant) => {
  //         if (!participant) {
  //           return;
  //         }
  //         const updatedStudies = participant.enrolled.filter((e) => e !== study.studyId) ?? [];
  //         const bodyObj = participant;
  //         bodyObj.enrolled = updatedStudies;
  //         await fetch(`http://localhost:5000/record/participant-edit/${participant.username}`, {
  //           method: 'POST',
  //           body: JSON.stringify(bodyObj),
  //           headers: {
  //             'Content-Type': 'application/json',
  //           },
  //         }).catch((e) => { window.alert(e); });
  //       },
  //     ),
  //   );
  // }

  // async function handleClose() {
  //   await getParticipants().then((participants) => { removeStudyForParticipants(participants); });
  //   const bodyObj = study;
  //   bodyObj.closed = '1';
  //   bodyObj.participants = [];
  //   await fetch('http://localhost:5000/add-study', {
  //     method: 'POST',
  //     body: JSON.stringify(bodyObj),
  //     headers: {
  //       'Content-Type': 'application/json',
  //     },
  //   }).catch((e) => { window.alert(e); });
  //   navigate('/researcher-home');
  // }

  return (
    <div>
      <div>EDIT STUDY: {study.title}</div>
      <div>
          <div>Title:</div>
          <input
            className="small-input"
            type="text"
            id="title"
            value={temp.title}
            onChange={updateTitle}
          />
          <div>Description:</div>
          <input
            className="small-input"
            type="text"
            id="description"
            value={temp.description}
            onChange={updateDescription}
          />
          <div>Compensation:</div>
          <input
            className="small-input"
            type="text"
            id="compensation"
            value={temp.compensation}
            onChange={updateCompensation}
          />
          <div>Duration:</div>
          <input
            className="small-input"
            type="text"
            id="duration"
            value={temp.duration}
            onChange={updateDuration}
          />
          <div>Lead Researcher:</div>
          <input
            className="small-input"
            type="text"
            id="researchers"
            value={temp.researchers}
            onChange={updateResearcher}
          />
        </div>
        <Button title="EDIT STUDY" type="submit" onPress={() => handleSubmit()}/>
        <Button title="DELETE STUDY" type="button" onPress={() => handleDelete()}/>
    </div>

    // <div className="Profile">
    //   <div className="profile-flex">
    //     <div className="header-left"> Edit Study </div>
    //     <div className="title-row">
    //       <div>Title</div>
    //       <input
    //         className="input-field"
    //         type="text"
    //         id="title"
    //         value={study.title}
    //         onChange={updateTitle}
    //       />
    //     </div>
    //     <div className="description-row">
    //       <div>Description</div>
    //       <input
    //         className="input-field"
    //         type="text"
    //         id="description"
    //         value={study.description}
    //         onChange={updateDescription}
    //       />
    //       <div>Compensation</div>
    //       <input
    //         className="input-field"
    //         type="text"
    //         id="compensation"
    //         value={study.compensation}
    //         onChange={updateCompensation}
    //       />
    //       <div>Duration</div>
    //       <input
    //         className="input-field"
    //         type="text"
    //         id="duration"
    //         value={study.duration}
    //         onChange={updateDuration}
    //       />
    //       <div>Lead Researcher</div>
    //       <input
    //         className="input-field"
    //         type="text"
    //         id="researchers"
    //         value={study.researchers}
    //         onChange={updateResearcher}
    //       />
    //     </div>
    //     <div className="profile-row">
    //       <div>Tags</div>
    //     </div>
    //     <div className="profile-row">
    //       <Select
    //         options={Tags}
    //         isMulti
    //         onChange={(tags) => updateTags(tags)}
    //         value={study.tags}
    //         styles={customStyles}
    //       />
    //     </div>
    //     <input className="signup-button" type="submit" value="Edit Study" onClick={handleSubmit} />
    //     <input className="signup-button" type="button" value="Delete Study" onClick={handleDelete} />
    //     <input className="signup-button" type="button" value="Close Study" onClick={handleClose} />
    //   </div>
    // </div>
  );
}

export default EditStudy;
