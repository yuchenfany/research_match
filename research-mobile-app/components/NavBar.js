/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-filename-extension */
import React, { useState } from 'react';
import { View, Button, Text, StyleSheet } from 'react-native';

function NavBar({user, setUser, navigation}) {
  const [accountToggled, setAccountToggled] = useState(false);
  const editProfile = () => {
    // store current state in case of 'Cancel'
    // localStorage.setItem(user.username, JSON.stringify(user));
    // navigate(`/${(user.type ?? 0) === 0 ? 'participant' : 'researcher'}-edit`);
  };

  const accountOptions = (
    <View
      className="acct-options"
      style={{ display: accountToggled ? 'flex' : 'none', flexDirection: 'row' }}
    >
      <Button
	    title="EDIT PROFILE"
        className="account-btn"
        onPress={editProfile}
      >
      </Button>
      <Button
	    title="DELETE ACCOUNT"
        className="account-btn"
        onPress={async () => { navigation.navigate('DeleteAccount', { user, setUser }); }}
      >
      </Button>
      <Button
	    title="LOGOUT"
        className="logout-btn"
        onPress={async () => { navigation.navigate('Research Application', { user, setUser }); }}
      >
      </Button>
    </View>
  );

  const toggleAccountOptions = () => {
    setAccountToggled(!accountToggled);
  };

  const studyListButton = (user.type ?? 0) ? <Button
      title="MY STUDIES"
      className="nav-btn"
      onPress={async () => {navigation.push('ParticipantStudies', { user, setUser });}}
    ></Button> : null;

  return (
    <View className="nav" style={{display: 'flex', flexDirection: 'row', marginBottom: '0.5rem'}}>
      <Button
	  	  title="HOME"
        className="nav-btn"
        onPress={async () => {
			navigation.push(
				`${(user.type ?? 0) === 0 ? 'Participant' : 'Researcher'}Home`,
				{ user, setUser }
			);
        }}
      >
      {studyListButton}
      </Button>
      <Button title="MESSAGES" className="nav-btn"></Button>
      <Button title="ACCOUNT" className="account-btn" onPress={toggleAccountOptions}></Button>
      {accountOptions}
    </View>
  );
}

export default NavBar;
