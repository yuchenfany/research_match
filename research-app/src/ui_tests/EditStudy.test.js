/**
 * @jest-environment jsdom
 */

/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-filename-extension */
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import React from 'react';
// import { jest } from '@jest/globals';

import EditStudy from '../components/EditStudy';
import NavBar from '../components/NavBar';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

test('Edit Study studies header', () => {
  const mockStudy = {
    title: 'title', description: 'description', duration: 0, leadResearcher: 'lead',
  };
  const mockSetStudy = jest.fn();

  render(
    <Router>
      <EditStudy study={mockStudy} setStudy={mockSetStudy} />
    </Router>,
  );
  const header = screen.getByText('Edit Study');
  expect(header).toBeInTheDocument();
});

test('snapshot test', () => {
  const mockStudy = {
    title: 'title', description: 'description', duration: 0, leadResearcher: 'lead',
  };
  const mockSetStudy = jest.fn();

  const component = renderer.create(
    <Router>
      <EditStudy study={mockStudy} setStudy={mockSetStudy} />
    </Router>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
