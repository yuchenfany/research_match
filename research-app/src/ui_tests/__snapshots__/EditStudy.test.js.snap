// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="Profile"
>
  <div
    className="profile-flex"
  >
    <div
      className="header-left"
    >
       Edit Study 
    </div>
    <div
      className="title-row"
    >
      <div>
        Title
      </div>
      <input
        className="input-field"
        id="title"
        onChange={[Function]}
        type="text"
        value="title"
      />
    </div>
    <div
      className="description-row"
    >
      <div>
        Description
      </div>
      <input
        className="input-field"
        id="description"
        onChange={[Function]}
        type="text"
        value="description"
      />
      <div>
        Compensation
      </div>
      <input
        className="input-field"
        id="compensation"
        onChange={[Function]}
        type="text"
      />
      <div>
        Duration
      </div>
      <input
        className="input-field"
        id="duration"
        onChange={[Function]}
        type="text"
        value={0}
      />
      <div>
        Lead Researcher
      </div>
      <input
        className="input-field"
        id="researchers"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="profile-row"
    >
      <div>
        Tags
      </div>
    </div>
    <div
      className="profile-row"
    >
      <div
        className=" css-b62m3t-container"
        onKeyDown={[Function]}
      >
        <span
          className="css-1f43avz-a11yText-A11yText"
          id="react-select-3-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className=" css-108aapi-control"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className=" css-319lph-ValueContainer"
          >
            <div
              className=" css-14el2xx-placeholder"
              id="react-select-3-placeholder"
            >
              Select...
            </div>
            <div
              className=" css-6j8wv5-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-controls="react-select-3-listbox"
                aria-describedby="react-select-3-placeholder"
                aria-expanded={false}
                aria-haspopup={true}
                aria-owns="react-select-3-listbox"
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className=""
                disabled={false}
                id="react-select-3-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "color": "inherit",
                    "font": "inherit",
                    "gridArea": "1 / 2",
                    "label": "input",
                    "margin": 0,
                    "minWidth": "2px",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "100%",
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className=" css-1hb7zxy-IndicatorsContainer"
          >
            <span
              className=" css-1okebmr-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              className=" css-tlfecz-indicatorContainer"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <svg
                aria-hidden="true"
                className="css-tj5bde-Svg"
                focusable="false"
                height={20}
                viewBox="0 0 20 20"
                width={20}
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <input
      className="signup-button"
      onClick={[Function]}
      type="submit"
      value="Update"
    />
    <input
      className="signup-button"
      onClick={[Function]}
      type="button"
      value="Delete Study"
    />
    <input
      className="signup-button"
      onClick={[Function]}
      type="button"
      value="Close Study"
    />
  </div>
</div>
`;
